src/org/gk/slicing/SlicingEngine.java

```diff
@@ -228,15 +294,196 @@ public void slice() throws Exception {
+		originalCompartment = (GKInstance) member.getAttributeValue(ReactomeJavaConstants.compartment);
+        member.setAttributeValue(ReactomeJavaConstants.compartment, cytoplasm);
+
+        // Control.
+		attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+        assertEquals(1, attrValue.size());
+
+        // Populate the compartments.
+        populateEntitySet(entity, ReactomeJavaConstants.compartment, null);
+        attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+        assertEquals(2, attrValue.size());
+        
+        // Remove the added compartment.
+        member.setAttributeValue(ReactomeJavaConstants.compartment, originalCompartment);
+        populateEntitySet(entity, ReactomeJavaConstants.compartment, null);
+        attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+		// System.out.println(attrValue);
+        assertEquals(1, attrValue.size());
+
+		/*
+		 * Changing the compartments of the containing entity sets.
+		 */
+    	// Each of these EntitySet's (and all of their respective members) have just one compartment.
+    	List<Long> dbIds = Arrays.asList(9619112L,  // ACEI pro-drugs [extracellular region]
+										 182588L,   // INK4 [cytolsol]
+										 3215203L); // ING [nucleoplasm]
+
+    	// Compartment not included in the EntitySet's above.
+    	GKInstance plasmaMembrane = testDBA.fetchInstance(876L);
+
+		for (Long dbId : dbIds) {
+			entity = testDBA.fetchInstance(dbId);
+			originalCompartment = (GKInstance) entity.getAttributeValue(ReactomeJavaConstants.compartment);
+
+    		// Control.
+			attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+    		assertEquals(1, attrValue.size());
+
+    		// Fill attribute values.
+			populateEntitySet(entity, ReactomeJavaConstants.compartment, null);
+			entity.addAttributeValue(ReactomeJavaConstants.compartment, plasmaMembrane);
+			attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+			System.out.println(attrValue);
+    		assertEquals(2, attrValue.size());
+
+    		// Remove added attribute values.
+			entity.setAttributeValue(ReactomeJavaConstants.compartment, originalCompartment);
+			populateEntitySet(entity, ReactomeJavaConstants.compartment, null);
+			attrValue = entity.getAttributeValuesList(ReactomeJavaConstants.compartment);
+    		assertEquals(1, attrValue.size());
+    	}
+    }
+    
+    /**
+     * 
+     * @param msg
+     * @param ps
+     */
+    static void logAndPrintln(String msg, PrintStream ps) {
+    	logger.info(msg);
+    	if (ps != null)
+			ps.println(msg);
+    }
+
+    /**
+     * Return true if at least one "compare" value is provided by user.
+     * 
+     * @param args
+     * @return 
+     * @return boolean (true if compare property is found, false otherwise).
+     * @throws IOException
+     */
+    private static boolean isCompareRequested(Properties properties) throws IOException {
```

Comment:
Please add a direct flag in the SlicingTool.prop, e.g., needCompareToPreviousSlice, to control this. This type of check is too clumsy :-(

Reply:
I think that is a good idea : )
Fixed in [5d0061c](https://github.com/reactome/CuratorTool/pull/20/commits/5d0061cf2e6db424efa382db7c7d1255250b379b).